x-pack/plugins/session_view/public/components/process_tree_alert/helpers.ts

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+const STATUS_TO_COLOR_MAP: Record<string, string> = {
```

Comment:
Since our main goal here is to stay in sync with the Status on the Alert Flyout, I'm wondering if we should instead reuse the same component that Alert Flyout uses to render the badge, I think it is the `<RuleStatus/>` component declared on `x-pack/plugins/security_solution/public/timelines/components/timeline/body/renderers/rule_status.tsx`.

But not sure if we could reuse that since `RuleStatus` is in security_solution, 

@andrew-goldstein do you know if we can / should reuse it or if it's better to manage on the `session_view` plugin, even if that means they can potentially become different from each other at a future point?


Alert Row in Session View:
![image](https://user-images.githubusercontent.com/19270322/157322721-7d70795c-9180-4bc9-9e50-f53f3dbdbeda.png)

Alert Flyout:
![image](https://user-images.githubusercontent.com/19270322/157322918-144579fe-c121-4996-8f94-a981da36891f.png)

Reply:
> @andrew-goldstein do you know if we can / should reuse it or if it's better to manage on the `session_view` plugin, even if that means they can potentially become different from each other at a future point?

Assuming the following goals:

✅  Avoiding the Session View plugin having a dependency on the Security Solution plugin
✅  Providing a non-interactive "default" rendering of the (`EuiBadge`) alert status badge, such that any Kibana app can consume the session viewer plugin
✅  For the Security Solution, still providing a consistent UX, in the form of an interactive status badge, so users can update the status throughout the app

Would you be willing to consider the following?

- Providing a default, non-interactive rendering of the alert status badge (via a similarly styled `EuiBadge`)
- Enable consumer apps to optionally override the default rendering of the alert status badge (via props)?

@kqualters-elastic may also have an opinion on this, but I think the above would achieve all of the goals above, and hopefully doesn't introduce too much complexity.

If providing the override would introduce too much complexity, it seems reasonable to unblock this PR by implementing a similarly styled non-interactive badge based on `EuiBadge`, and then adding the optional prop for apps to override it in a follow-up PR.

Reply:
alternatively you could ensure that they don't diverge too far by using shared types, if they are `export type Whatever` and `import type Whatever` they get compile time erased and so don't add to bundle size I believe

Reply:
I like @kqualters-elastic suggestion of using shared types, I'm not sure if that's how we share types, but I see we can import `STATUS_VALUES` from `@kbn/rule-data-utils`, maybe we could add a type like `STATUS_VALUES_TO_COLOR_MAP` to `@kbn/rule-data-utils` and update both places to use it so we can get in sync?

@kbn/rule-data-utils:

```js
export const enum STATUS_VALUES_TO_COLOR_MAP {
    open = 'primary',
    acknowledged = 'warning',
    closed = 'default',
  }
```

and update on components to use it:

```js
import { STATUS_VALUES_TO_COLOR_MAP } from '@kbn/rule-data-utils';

const mapping = {
  open: STATUS_VALUES_TO_COLOR_MAP.open,
  acknowledged: STATUS_VALUES_TO_COLOR_MAP.acknowledged,
  closed: STATUS_VALUES_TO_COLOR_MAP.closed,
};
```

or maybe even exporting the const directly could be more beneficial?

```js
export const STATUS_VALUES_TO_COLOR_MAP = {
  open: 'primary',
  acknowledged: 'warning',
  closed: 'default',
} as const;
```

Reply:
Ya a lot of things with types would probably work. Passing in a component to override for the badge though would have the benefit of being able to change the alert status from directly in session view, and letting users do that for alerts that could be pages and pages apart on the alerts page or not at all in the query of the current table would be a relatively easy way to make full use of the entire tree view. Plus this would make use of existing security solution permissions logic, for users who can see the status of an alert but can't change it, etc.

Reply:
Thanks for input folks! After some more discussion we decided to address this in a separate PR, when we add the implementation of alerts fly out